src/main/scala/Task3/package.scala

```diff
@@ -0,0 +1,97 @@
+package object Task3 {
+
+  case class Person(name: String, age: Int)
+
+  def dir(x: AnyRef): Unit = println(x.getClass.getMethods.map(_.getName).distinct.sorted.mkString(" "))
+
+  trait Validator[T] {
+    /**
+      * Validates the value.
+      *
+      * @param value value to be validated.
+      * @return Right(value) in case the value is valid, Left(message) on invalid value
+      */
+    def validate(value: T): Either[String, T]
+
+    /**
+      * And combinator.
+      *
+      * @param other validator to be combined with 'and' with this validator.
+      * @return the Right(value) only in case this validator and <code>other</code> validator returns valid value,
+      *         otherwise Left with error messages from the validator that failed.
+      */
+    def and(other: Validator[T]): Validator[T] = new Validator[T] {
+      override def validate(value: T): Either[String, T] = {
+        Validator.this.validate(value) match {
+          case Right(v) => other.validate(v)
+          case Left(error) => Left(error)
+        }
+      }
+    }
+    /**
+      * Or combinator.
+      *
+      * @param other validator to be combined with 'or' with this validator.
+      * @return the Right(value) only in case either this validator or <code>other</code> validator returns valid value,
+      *         otherwise Left with error messages from both validators.
+      */
+    def or(other: Validator[T]): Validator[T] = new Validator[T] {
+      override def validate(value: T): Either[String, T] = {
+        Validator.this.validate(value) match {
+          case Right(v) => Right(v)
+          case Left(_) => other.validate(value)
+        }
+      }
+    }
+  }
+
+  object Validator {
+
+    implicit class Validated[T](t: T) {
+      def validate(validator: Validator[T] = new Validator[T] {
+        override def validate(value: T): Either[String, T] = Right(value)
+      }): Either[String, T] = {
```

Comment:
this is confusing, you provide default validator that is always valid.
what you need to do is:
```scala
implicit class Validated[T](t: T) {
      def validate(implicit validator: Validator[T]): Either[String, T] = validator.validate(t)
```
having this allows you either to not pass validator at all (it will be searched by compiler as it is implicit), or you can explicitly pass it.